x-pack/plugins/security_solution/public/entity_analytics/hooks/use_risk_contributing_alerts.ts

```diff
@@ -6,66 +6,97 @@
  */
 
 import { useEffect } from 'react';
+import type { ALERT_RULE_NAME, ALERT_RULE_UUID } from '@kbn/rule-data-utils';
+import type { EntityRiskInput } from '../../../common/entity_analytics/risk_engine';
+import { RiskScoreEntity } from '../../../common/entity_analytics/risk_engine';
 import { useQueryAlerts } from '../../detections/containers/detection_engine/alerts/use_query';
 import { ALERTS_QUERY_NAMES } from '../../detections/containers/detection_engine/alerts/constants';
 
 import type {
   UserRiskScore,
   HostRiskScore,
 } from '../../../common/search_strategy/security_solution/risk_score/all';
-import { getAlertsQueryForRiskScore } from '../common/get_alerts_query_for_risk_score';
-
-import { useRiskEngineSettings } from '../api/hooks/use_risk_engine_settings';
+import { isUserRiskScore } from '../../../common/search_strategy/security_solution/risk_score/all';
 
 interface UseRiskContributingAlerts {
   riskScore: UserRiskScore | HostRiskScore | undefined;
-  fields?: string[];
+  entityType: RiskScoreEntity;
+}
+
+interface AlertData {
+  [ALERT_RULE_UUID]: string;
+  [ALERT_RULE_NAME]: string;
 }
 
-interface Hit {
-  fields: Record<string, string[]>;
+interface AlertHit {
+  _id: string;
   _index: string;
+  _source: AlertData;
+}
+
+export interface InputAlert {
+  alert: AlertData;
+  input: EntityRiskInput;
   _id: string;
 }
 
-interface UseRiskContributingAlertsResult {
+export interface UseRiskContributingAlertsResult {
   loading: boolean;
   error: boolean;
-  data?: Hit[];
+  data?: InputAlert[];
 }
 
 /**
  * Fetches alerts related to the risk score
  */
 export const useRiskContributingAlerts = ({
   riskScore,
-  fields,
+  entityType,
 }: UseRiskContributingAlerts): UseRiskContributingAlertsResult => {
-  const { data: riskEngineSettings } = useRiskEngineSettings();
-
-  const { loading, data, setQuery } = useQueryAlerts<Hit, unknown>({
-    // is empty query, to skip fetching alert, until we have risk engine settings
+  const { loading, data, setQuery } = useQueryAlerts<AlertHit, unknown>({
     query: {},
     queryName: ALERTS_QUERY_NAMES.BY_ID,
   });
 
   useEffect(() => {
-    if (!riskEngineSettings?.range?.start || !riskScore) return;
-
-    setQuery(
-      getAlertsQueryForRiskScore({
-        riskRangeStart: riskEngineSettings.range.start,
-        riskScore,
-        fields,
-      })
-    );
-  }, [setQuery, riskScore, riskEngineSettings?.range?.start, fields]);
+    if (!riskScore) return;
+    setQuery({
+      query: {
+        ids: {
+          values: getInputs(riskScore).map((input) => input.id),
+        },
+      },
+    });
+  }, [riskScore, setQuery]);
 
   const error = !loading && data === undefined;
 
+  const inputs =
```

Comment:
can we use `getInputs` here like on line 66? Maybe we could define a variable for the inputs and they could share